src/imitation/util/logger.py

```diff
@@ -0,0 +1,200 @@
+import contextlib
+import os
+from typing import ContextManager, Optional, Sequence
+
+import stable_baselines.logger as sb_logger
+
+
+def _build_output_formats(folder: str,
+                          format_strs: Sequence[str] = None,
+                          ) -> Sequence[sb_logger.KVWriter]:
+  """Build output formats for initializing a Stable Baselines Logger.
+
+  Args:
+    folder: Path to directory that logs are written to.
+    format_strs: An list of output format strings. For details on available
+      output formats see `stable_baselines.logger.make_output_format`.
+  """
+  os.makedirs(folder, exist_ok=True)
+  output_formats = [sb_logger.make_output_format(f, folder)
+                    for f in format_strs]
+  return output_formats
+
+
+class _AccumulatingLogger(sb_logger.Logger):
+
+  def __init__(self,
+               format_strs: Sequence[str],
+               *,
+               mean_logger: sb_logger.Logger,
+               subdir: str):
+    """Like Logger, except also accumulates logkv_mean on the mean_logger.
+
+    Args:
+      format_strs: An list of output format strings. For details on available
+        output formats see `stable_baselines.logger.make_output_format`.
+      mean_logger: A background logger that is used to keep track of log
+        means.
+      subdir: Used to build the logging directory. Also used in the logging
+        prefix for every key written to both this logger and `mean_logger`.
+    """
+    self.subdir = subdir
+    self.mean_logger = mean_logger
+    folder = os.path.join(self.mean_logger.dir, "raw", subdir)
+    output_formats = _build_output_formats(folder, format_strs)
+    os.makedirs(folder, exist_ok=True)
+    super().__init__(folder, output_formats)
+
+  def logkv(self, key, val):
+    raw_key = os.path.join("raw", self.subdir, key)
+    super().logkv(raw_key, val)
+
+    mean_key = os.path.join("mean", self.subdir, key)
```

Comment:
Is the `"mean"` prefix here necessary? Won't everything in this logger be means? Or can it also be logged to directly, in which case maybe consider renaming to `root_logger` or `parent_logger`?

Reply:
The `_AccumulatingLogger` only uses this logger instance to log means, so the attribute name `self.mean_logger` within `_AccumulatingLogger` seems appropriate to me

Reply:
> Is the "mean" prefix here necessary? 

Discussion in [answer to previous comment](https://github.com/HumanCompatibleAI/imitation/pull/135#discussion_r362684318).